plugin/trino-hive/src/test/java/io/trino/plugin/hive/ion/IonPageSourceSmokeTest.java

```diff
@@ -52,19 +59,28 @@
 import static io.trino.plugin.hive.HiveTestUtils.projectedColumn;
 import static io.trino.plugin.hive.HiveTestUtils.toHiveBaseColumnHandle;
 import static io.trino.plugin.hive.acid.AcidTransaction.NO_ACID_TRANSACTION;
+import static io.trino.plugin.hive.ion.IonWriterOptions.BINARY_ENCODING;
+import static io.trino.plugin.hive.ion.IonWriterOptions.ION_ENCODING_PROPERTY;
+import static io.trino.plugin.hive.ion.IonWriterOptions.TEXT_ENCODING;
 import static io.trino.plugin.hive.util.SerdeConstants.LIST_COLUMNS;
 import static io.trino.plugin.hive.util.SerdeConstants.LIST_COLUMN_TYPES;
 import static io.trino.spi.type.BooleanType.BOOLEAN;
 import static io.trino.spi.type.IntegerType.INTEGER;
 import static io.trino.spi.type.RowType.field;
 import static io.trino.spi.type.VarcharType.VARCHAR;
 import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.util.stream.Collectors.toList;
 
 /**
  * Most basic test to reflect PageSource-fu is wired up correctly.
  */
 public class IonPageSourceSmokeTest
 {
+    // In the Ion binary format, a value stream is always start with binary version marker. This help distinguish Ion binary
+    // data from other formats, including Ion text format.
+    private static final byte[] BINARY_VERSION_MARKER = {(byte) 0xE0, (byte) 0x01, (byte) 0x00, (byte) 0XEA};
+    private static final String EXPECTED_TEXT = "{foo:3,bar:6}";
```

Comment:
This is OK as it is, no need to change it, just food for thought: I probably would just assert on "starts with text opening brace", anything else gets brittle fast. This is not so bad in and of itself though.